core/amber/src/main/scala/edu/uci/ics/amber/engine/common/ambertag/neo/Identifier.scala

```diff
@@ -0,0 +1,17 @@
+package edu.uci.ics.amber.engine.common.ambertag.neo
+
+trait Identifier
+
+object Identifier {
+
+  case class EmptyIdentifier() extends Identifier
+  case class ActorIdentifier(name: String) extends Identifier
+  case class ClientIdentifier() extends Identifier
+  case class ControllerIdentifier() extends Identifier
+  case class SelfIdentifier() extends Identifier
```

Comment:
do we plan to use these in the code later? Specifically about Self, Empty and Client

Reply:
I have seen this way of putting case class inside an object multiple times. Is it a standard way to do this? It kind of confuses me because an object is supposed to be singleton, but we obviously have multiple instances of these case classes.

Reply:
Yes. We don't want to use `self` and `context.parent` anymore because they're `ActorRef`s. I've already used `Identifier.Self` in DataProcessor class to notify the main thread of processing status. I want to make sure messages always go through our FIFO/exactly-once protocol.

Reply:
why have different types of identifier? They can all just be Identifier (string: name)?

Reply:
I have been thinking about a better name. The best I have been able to come up with is VirtualIdentity